lib/ignore/ignore.go

```diff
@@ -55,26 +69,26 @@ func New(withCache bool) *Matcher {
 	return m
 }
 
-func (m *Matcher) Load(file string) error {
+func (m *Matcher) Load(file string, includePath string) error {
```

Comment:
Yes, recursively.
Initially, includePath is the folderPath. This allows `#include myignores.txt` from the new `.syncthing/ignore.txt` file, hence providing backward compatibility.
Subsequent includePaths are relative to the path of the ignore file currently being parsed.